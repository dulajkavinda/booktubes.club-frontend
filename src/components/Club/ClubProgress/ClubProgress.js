import React from "react";

import styles from "./ClubProgress.module.css";
import { Progress } from '@chakra-ui/react';
import { Stack, VStack } from '@chakra-ui/react';
import { Heading } from '@chakra-ui/react';

export default function ClubProgress({users}) {

    const data = {
        "members": [
            {
                "userid": "001",
                "userName": "John Doe",
                "percentage": 24.00
            },
            {
                "userid": "0zYPmgHgOHWWt0X12iudnT0bNZ03",
                "userName": "Emma MacCay",
                "percentage": 52.10
            },
            {
                "userid": "003",
                "userName": "Beth Johns",
                "percentage": 73.00
            },
            {
                "userid": "004",
                "userName": "Ross Geller",
                "percentage": 24.00
            },
            {
                "userid": "005",
                "userName": "Emily Blunt",
                "percentage": 52.10
            },
            {
                "userid": "006",
                "userName": "Seth Myers",
                "percentage": 100
            }
        ]
    };

    return (
        <div className={styles.memberProgressModal}>
            <Heading className={styles.progressHeading} size='md'>Members' Progress</Heading>
            <VStack spacing={4}>
                {data.members.map((data) => (
                    <Stack>
                        <p className={styles.pName}>{data.userName}</p>
                        <Progress size='xs' value={data.percentage} />
                    </Stack>
                ))}
            </VStack>

        </div>
    );
}