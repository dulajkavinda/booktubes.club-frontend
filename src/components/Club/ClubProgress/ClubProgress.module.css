.memberProgressModal {
    padding: 10px 20px;
    max-height: 200px;
    overflow-y: scroll;
}

.progressBar {
    max-width: 80px;
}

.pName {
    width: 200px;
}

.progressHeading {
    padding-bottom: 20px;
}
