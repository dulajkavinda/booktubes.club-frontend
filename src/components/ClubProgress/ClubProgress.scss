.member-progress-modal {
    max-height: 400px;
    overflow-y: scroll;

    th {
        width: 50%;
    }
    td {
    }
}

.progress-bar {
    max-width: 80px;
}
