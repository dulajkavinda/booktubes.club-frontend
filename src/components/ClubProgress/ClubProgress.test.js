import React from "react";
import { shallow, mount } from "enzyme";
import ClubProgress from "./ClubProgress";

let wrapper;
beforeEach(() => {
  wrapper = shallow(<ClubProgress />);
});

describe("dashboard basic tests", () => {
  it("is snapshot changed", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
