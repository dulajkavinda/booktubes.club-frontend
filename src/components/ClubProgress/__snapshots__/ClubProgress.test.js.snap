// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dashboard basic tests is snapshot changed 1`] = `
<div
  className="member-progress-modal"
>
  <ForwardRef(Table)
    aria-label="Member progress table"
    sx={
      Object {
        "minWidth": 350,
      }
    }
  >
    <ForwardRef(TableBody)>
      <ForwardRef(TableRow)
        key="001"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0zYPmgHgOHWWt0X12iudnT0bNZ03"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="003"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="004"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="005"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="006"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="007"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="008"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="009"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0012"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0013"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0033"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0015"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0028"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0037"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0019"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          John Doe
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={24}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0020"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Emma MaCay
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={52.1}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
      <ForwardRef(TableRow)
        key="0030"
        sx={
          Object {
            "&:last-child td, &:last-child th": Object {
              "border": 0,
            },
          }
        }
      >
        <ForwardRef(TableCell)
          component="th"
          scope="row"
        >
          Beth Johns
        </ForwardRef(TableCell)>
        <ForwardRef(TableCell)>
          <ForwardRef(LinearProgress)
            className="progress-bar"
            value={73}
            variant="determinate"
          />
        </ForwardRef(TableCell)>
      </ForwardRef(TableRow)>
    </ForwardRef(TableBody)>
  </ForwardRef(Table)>
</div>
`;
